src/backend/access/external/pxfutils.c

```diff
@@ -133,3 +137,60 @@ static void process_request(ClientContext* client_context, char *uri)
 
 
 }
+
+/*
+ * Finds ip address of any available loopback interface(ipv4/ipv6).
+ * Returns ip for ipv4 addresses and [ip] for ipv6 addresses.
+ *
+ */
+char* get_loopback_ip_addr() {
+	struct ifaddrs *ifaddr, *ifa;
+	int family, s, n;
+	char host[NI_MAXHOST];
+	char *loopback_addr;
+
+	if (getifaddrs(&ifaddr) == -1) {
+		elog(ERROR, "Unable to obtain list of network interfaces.");
+	}
+
+	for (ifa = ifaddr, n = 0; ifa != NULL; ifa = ifa->ifa_next, n++) {
+		if (ifa->ifa_addr == NULL)
+			continue;
+
+		family = ifa->ifa_addr->sa_family;
+
+		if (family == AF_INET || family == AF_INET6) {
+			s = getnameinfo(ifa->ifa_addr,
+					(family == AF_INET) ?
+							sizeof(struct sockaddr_in) :
+							sizeof(struct sockaddr_in6), host, NI_MAXHOST,
+					NULL, 0, NI_NUMERICHOST);
+			if (s != 0) {
+				elog(WARNING, "Unable to get name information for interface, getnameinfo() failed: %s\n", gai_strerror(s));
+			}
+
+			//get loopback interface
+			if (ifa->ifa_flags & IFF_LOOPBACK) {
+				if (family == AF_INET)
+				{
+					loopback_addr = palloc(strlen(host) + 1);
+					sprintf(loopback_addr, "%s", host);
+				}
+				else
+				{
+					loopback_addr = palloc(strlen(host) + 3);
+					sprintf(loopback_addr, "[%s]", host);
+				}
+				break;
+			}
+		}
+	}
+
+	freeifaddrs(ifaddr);
+
+	if (loopback_addr == NULL)
+		elog(ERROR, "Unable to get loop back address");
```

Comment:
If function returned null it means there is no loopback interfaces available, and even if we return "127.0.0.1" no interface could handle requests to PXF.